### PR TITLE
Quick removal of more blocking in Sockets

### PR DESCRIPTION
**Pull request by Drawaes**
Found a group of tests that use a mix of async and APM but the entire test can be made non blocking. Hopefully removing these is making the outerloop more stable/worth it :)

Ref #25346

/cc @stephentoub @davidsh

### CODE REVIEW

**Comment by Drawaes** [comment]
@dotnet-bot Test outerloop Linux x64 Release Build please
@dotnet-bot Test outerloop Windows x64 Release Build please
@dotnet-bot Test outerloop OSX x64 Release Build please

File: src/System.Net.Sockets/tests/FunctionalTests/AgnosticListenerTest.cs [hunk 1]
```diff
@@ -33,59 +33,56 @@ public void Create_Success()
...
-            IAsyncResult asyncResult = listener.BeginAcceptTcpClient(null, null);
+            TcpListener listener = SocketTestExtensions.CreateAndStartTcpListenerOnAnonymousPort(out int port);
+            Task<TcpClient> acceptTask = Task.Factory.FromAsync(listener.BeginAcceptTcpClient(null, null), listener.EndAcceptTcpClient);
 
             TcpClient client = new TcpClient(AddressFamily.InterNetwork);
-            client.ConnectAsync(IPAddress.Loopback, port).GetAwaiter().GetResult();
+            await client.ConnectAsync(IPAddress.Loopback, port);
 
-            TcpClient acceptedClient = listener.EndAcceptTcpClient(asyncResult);
+            TcpClient acceptedClient = await acceptTask;
```
Comment (stephentoub) [inline src/System.Net.Sockets/tests/FunctionalTests/AgnosticListenerTest.cs]: Nit: It'd be better to use the WhenAllOrAnyFailed helper here with the ConnectAsync and acceptTasks.

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop Windows x64 Release Build please

**Comment by Drawaes** [comment]
I made the change to the WhenAllOrFail, and then used the website to do a merge (bad idea) so have now fixed.

**Comment by Drawaes** [comment]
Do we want another set of outerloops on this?

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build please
@dotnet-bot test Outerloop Linux x64 Debug Build please

**Comment by Drawaes** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build please
@dotnet-bot test Outerloop Linux x64 Debug Build please

**Comment by Drawaes** [comment]
Windows had a timeout, (10 seconds for a loopback test on nano) so I will give it a shot again, if it fails we can try ramming the timeout up if you want but I think 10 seconds shows there is more work to be done

@dotnet-bot test Outerloop Windows x64 Debug Build please

**Comment by stephentoub** [comment]
> Windows had a timeout

Which test? Some of the connect tests have been frequently failing... seems like something else is going on.

**Comment by Drawaes** [comment]
It was a connect for IPV6 from Nano only (all other windows and Linux passed) on loopback. It hit the timeout of 10 seconds (there wasn't one before).

**Comment by Drawaes** [comment]
Passed that time. So it was just a timeout being hit, I think there are still blocking gremlins in these tests... I will plug on with eliminating them.

**Comment by stephentoub** [comment]
> there wasn't one before

When was it added?

**Comment by Drawaes** [comment]
This PR
https://github.com/dotnet/corefx/pull/25460/files#diff-15b23d64da42da478b7d926f7782c59eR61

**Comment by stephentoub** [comment]
Ok. Please remove it then.

**Comment by Drawaes** [comment]
I sure can and will....

but I would also say ... 10 seconds for a loop back connect test? eh? its a fail in my book . (merely registering my protest 😆 ).

**Comment by stephentoub** [comment]
> 10 seconds for a loop back connect test? eh? its a fail in my book

I completely agree. But adding a test that causes Outerloop to explicitly fail because of it only makes things worse until the real root cause is found. I find it very hard to believe it's just because of the blocking in other tests.

**Comment by Drawaes** [comment]
I tend to agree, it has a similar behavior (random OS/Distro in async tests, timeouts being hit). But I never investigated any area other than Net.Security in depth. Still, worth removing as much blocking as possible to eliminate it.

(Plus its a good, late night task when I don't want to think too hard ... :) )

**Comment by Drawaes** [comment]
@dotnet-bot Test outerloop Linux x64 Release Build please
@dotnet-bot Test outerloop Windows x64 Release Build please

**Comment by Drawaes** [comment]
Was a System.Drawing failure....

```
17:18:28 D:\j\workspace\windows-TGrou---2a8f9c29\Tools\tests.targets(484,5): warning :    System.Drawing.Common.Tests  Total: 4504, Errors: 0, Failed: 1, Skipped: 0, Time: 11.522s [D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Drawing.Common\tests\System.Drawing.Common.Tests.csproj]
17:18:28 D:\j\workspace\windows-TGrou---2a8f9c29\Tools\tests.targets(484,5): warning MSB3073: The command "D:\j\workspace\windows-TGrou---2a8f9c29\bin/tests/System.Drawing.Common.Tests/netfx-Windows_NT-Release-x86//RunTests.cmd D:\j\workspace\windows-TGrou---2a8f9c29\bin/testhost/netfx-Windows_NT-Release-x86/" exited with code 1. [D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Drawing.Common\tests\System.Drawing.Common.Tests.csproj]
17:18:28 D:\j\workspace\windows-TGrou---2a8f9c29\Tools\tests.targets(492,5): error : One or more tests failed while running tests from 'System.Drawing.Common.Tests' please check D:\j\workspace\windows-TGrou---2a8f9c29\bin/tests/System.Drawing.Common.Tests/netfx-Windows_NT-Release-x86/testResults.xml for details! [D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Drawing.Common\tests\System.Drawing.Common.Tests.csproj]
17:18:28 D:\j\workspace\windows-TGrou---2a8f9c29\dir.traversal.targets(77,5): error : (No message specified) [D:\j\workspace\windows-TGrou---2a8f9c29\src\tests.builds]
```

@dotnet-bot Test NETFX x86 Release Build please

**Comment by benaadams** [comment]
@dotnet-bot Test outerloop Windows x64 Release Build please

**Comment by stephentoub** [comment]
@Drawaes, tests continue to fail with the timeouts that have been added.

**Comment by Drawaes** [comment]
Add okay, I was confused I took all the timeouts out, but the timeout that is failing is from a previous PR, I can remove it in this PR for "stability".... still worried about that 10seconds :P But I have a bunch of other blocks I can attack.

**Comment by wfurt** [comment]
This generally looks ok to me. But are we loosing test coverage for synchronous calls? 
I would feel we still should be ability to test both synchronous and blocking calls in reasonable way.

**Comment by Drawaes** [comment]
I haven't removed any sync calls. Just the APM and async await calls I removed the blocking from.

**Comment by Drawaes** [comment]
@dotnet-bot Test outerloop Windows x64 Release Build please

File: src/Common/tests/System/Net/Sockets/TestSettings.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@ namespace System.Net.Sockets.Tests
     public static class TestSettings
     {
         // Timeout values in milliseconds.
-        public const int PassingTestTimeout = 10000;
+        public const int PassingTestTimeout = 30000;
```
Comment (stephentoub) [inline src/Common/tests/System/Net/Sockets/TestSettings.cs]: This has been at 10s for months, no?  But it only started causing problems when you added timeouts to some tests in a previous PR?  If so, please leave this as it was and remove those added timeouts.

Comment (Drawaes) [reply src/Common/tests/System/Net/Sockets/TestSettings.cs]: Agreed, I planned on rolling back the timeouts, this was just for my info before I did it.

**Comment by Drawaes** [comment]
@dotnet-bot Test outerloop Linux x64 Release Build please
@dotnet-bot Test outerloop Windows x64 Release Build please
@dotnet-bot Test outerloop OSX x64 Release Build please

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build please
@dotnet-bot test Outerloop Windows x64 Release Build please

**Comment by stephentoub** [comment]
Thanks, @Drawaes.

**Pull request merged**